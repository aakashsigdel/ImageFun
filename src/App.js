import React, { Component } from 'react';
import { compose, map, prop } from 'ramda';

import ImagePicker from './components/ImagePicker';
import Image from './components/Image';
import AvailableActions from './components/AvailableActions';
import AppliedActions from './components/AppliedActions';

import './App.scss';

class App extends Component {
  state = {
    imageURL: null,
    availableActions: [
      {id: 1, action: 'rotate', name: 'Rotate'},
      {id: 2, action: 'translateX', name: 'Translate'},
      {id: 3, action: 'opacity', name: 'Opacity'},
      {id: 4, action: 'bounce', name: 'Bounce'}
    ],
    appliedActions: []
  }

  setImageURL = URLString => {
    this.setState({
      imageURL: URLString
    });
  }

  pushAvailableAction = action => {
    this.setState({
      appliedActions: [
        action,
        ...this.state.appliedActions
      ],
      availableActions: this.state.availableActions.filter(availAction => availAction.id !== action.id)
    });
  }

  popAppliedAction = action => {
    this.setState({
      appliedActions : this.state.appliedActions.filter(appliedActions => action.id !== appliedActions.id),
      availableActions: [
        action,
        ...this.state.availableActions
      ]
    });
  }

  render () {
    return (
      <div className="container">
        <Image
          imageURL={this.state.imageURL}
          actions={map(prop('action'))(this.state.appliedActions).join(' ')}
        />
        <ImagePicker setImageURL={this.setImageURL} />
        <AvailableActions
          availableActions={this.state.availableActions}
          pushAction={this.pushAvailableAction}
        />
        <AppliedActions
          appliedActions={this.state.appliedActions}
          popAction={this.popAppliedAction}
        />
      </div>
    );
  }
}

export default App;
