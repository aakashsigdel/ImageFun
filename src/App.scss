@import './_variables.scss';

body {
  font-family: 'Roboto', sans-serif;
  font-size: 16px;
  font-weight: 400;
  color: $primary_font;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  text-rendering: optimizeLegibility;
}

.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid lightgrey;
}

.actions {
  display: flex;
}

@media (max-width: 600px) {
  .actions {
    display: block;
  }
}
