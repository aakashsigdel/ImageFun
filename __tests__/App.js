import React from 'react';
import App from '../src/App';

import renderer from 'react-test-renderer';

describe('<App /> component', () => {
  it('should render <App /> correctly', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

