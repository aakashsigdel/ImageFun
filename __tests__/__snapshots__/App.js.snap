exports[`<App /> component should render <App /> correctly 1`] = `
<div
  className="container">
  <div
    className="image-container">
    <img
      className="image "
      src="test-file-stub" />
  </div>
  <input
    className="image-chooser"
    onChange={[Function]}
    type="file" />
  <div
    className="actions">
    <div
      className="available-actions-container">
      <div
        className="menu-title">
        Available Actions
      </div>
      <ul>
        <li
          className="action"
          onClick={[Function]}>
          <a>
            Rotate
          </a>
        </li>
        <li
          className="action"
          onClick={[Function]}>
          <a>
            Translate
          </a>
        </li>
        <li
          className="action"
          onClick={[Function]}>
          <a>
            Opacity
          </a>
        </li>
        <li
          className="action"
          onClick={[Function]}>
          <a>
            Bounce
          </a>
        </li>
      </ul>
    </div>
    <div
      className="applied-actions-container">
      <div
        className="menu-title">
        Applied Actions
      </div>
      <ul />
    </div>
  </div>
</div>
`;
